Interop/Runtime/src/jvm/kotlin/kotlinx/cinterop/JvmNativeMem.kt

```diff
@@ -95,19 +96,20 @@ internal object nativeMemUtils {
         return unsafe.allocateInstance(T::class.java) as T
     }
 
-    fun alloc(size: Long, align: Int): NativePointed {
-        val address = unsafe.allocateMemory(
-                if (size == 0L) 1L else size // It is a hack: `sun.misc.Unsafe` can't allocate zero bytes
-        )
-
+    fun allocRaw(size: Long, align: Int): NativePtr {
+        val address = unsafe.allocateMemory(size)
         if (address % align != 0L) TODO(align.toString())
-        return interpretOpaquePointed(address)
+        return address
     }
 
-    fun free(mem: NativePtr) {
+    fun freeRaw(mem: NativePtr) {
         unsafe.freeMemory(mem)
     }
 
+    fun alloc(size: Long, align: Int) = interpretOpaquePointed(NativeMemoryAllocator.alloc(size, align))
+
+    fun free(mem: NativePtr) = NativeMemoryAllocator.free(mem)
```

Comment:
So the allocator is not thread-safe anymore. Might be dangerous when using the compiler in parallel in Gradle daemon.